### PR TITLE
Solutions to series.py, fizzbuzz, and print grid

### PR DESCRIPTION
**Pull request by Eowyn42**
I've explored a few different ways to solve the print grid problem. Ultimately I realized there are symmetries - each line is constructed the exact same way except some of them use plus and minus characters and some use dash and space characters. 

Once I had solved that problem, I returned to the first couple of problems using this approach, except now the first argument was gone / fixed equal to 2, and the second argument is something like (n-1)/2. ie n =3 is the same shape as ndim = 2, ndash = 1 ; n = 15 is the same shape as ndim = 2, ndash = 7. But I'm having problems with the n = 11 case because it looks like the same shape as ndim = 2 and ndash = 4 but my solution produces ndim = 2, ndash = 5. I cribbed/tested/used a little rounding function from the internet, but neither rounding nor straight // truncation produced the desired result. I'm not sure if there's something I'm missing in the Part 2 part of the problem. . . my solution for Part 3 can solve Part 1 and MOST of Part 2, just not the n = 11 case.

### CODE REVIEW

**Comment by PythonCHB** [comment]
Wow! nice thorough job! 

I"ll go look at the code now....

File: students/eowyn/session02/print_grid.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+''' Construct grid that is composed of ndim boxes in 
+    either direction, and each box is composed of ndash 
+    lines both horizontally and vertically '''
+
+### First method uses global vars and functions for lines and verts
+### then a caller function to print it all
+
+# define global varaibles
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:10]: In general, global variables are considered "bad form" -- but this is a fine use case --essentially constants.

It is considerd good style to use ALL CAPS for contants, so you may want to use names like:

PLUS = "+"
MINUS = "-"

and so on. But no biggie.

File: students/eowyn/session02/print_grid.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+plus = "+"
+minus = "-"
+space = " "
+dash = '|'
+
+def make_line(ndim,ndash):
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:16]: another style point -- better to put a space after a comma: make_line(ndim, ndash).

File: students/eowyn/session02/print_grid.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    line = ''
+    for n in range(ndim):
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:18]: A loop is fine, but you could get the same effect with multiplication:

line = ndim * (plus + ndash*minus)

File: students/eowyn/session02/print_grid.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+def print_grid1(ndim,ndash):
+    for i in range(ndim):
+        print(make_line(ndim,ndash))
+        for j in range(ndash):
+            print(make_vert(ndim,ndash))
+    print(make_line(ndim,ndash))
+
+### Second method uses just one general construction function
+### and a caller function to print it all
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:38]: nice! much better to put all the printing in one place.

File: students/eowyn/session02/print_grid.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    for i in range(ndim):
+        print(make_row(ndim,ndash,'+','-'))
+        for j in range(ndash):
+            print(make_row(ndim,ndash,'|',' '))
+    print(make_row(ndim,ndash,'+','-'))
+
+### Now try the simple case where ndim is always 2
+### This is not producing the expected result for n = 11
+
+def roundno(no):
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:57]: python does have a "round" function....

round(n)

or 

round (n/2) ....

File: students/eowyn/session02/series.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+"""Return fibbonaci,lucas, and generic summed series series"""
+
+def fib(n):
+    """ Print the nth element of the fibbonaci sequence"""
+    return sum_series(n)
+
+def lucas(n):
```
Comment (PythonCHB) [inline students/eowyn/session02/series.py:8]: nice! once you've written the generic one -- use it to provide the others.

File: students/eowyn/session02/series.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    """ Print the nth element of the lucas sequence"""
+    return sum_series(n,2,1)
+
+def sum_series(n,n0=0,n1=1):
```
Comment (PythonCHB) [inline students/eowyn/session02/series.py:12]: perfect! and nice tests.

File: students/eowyn/session02/print_grid.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+''' Construct grid that is composed of ndim boxes in
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:3]: nice docstring -- but better style to put the quotes on their own line.

```
"""
Construct grid that is composed of ndim boxes in 
either direction, and each box is composed of ndash 
lines both horizontally and vertically
 """
```

File: students/eowyn/session02/print_grid.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+def print_grid(ndim,ndash):
+    for i in range(ndim):
+        print(make_row(ndim,ndash,'+','-'))
+        for j in range(ndash):
+            print(make_row(ndim,ndash,'|',' '))
+    print(make_row(ndim,ndash,'+','-'))
+
+### Now try the simple case where ndim is always 2
+### This is not producing the expected result for n = 11
```
Comment (PythonCHB) [inline students/eowyn/session02/print_grid.py:55]: it seems to work fine for me for n == 11 ? what is the issue?

**Review by PythonCHB — APPROVED** [review]
Very nice job!

**Comment by Eowyn42** [comment]
Hi Chris,

Thank you for the code review. The n=11 issue is that the image on the
problem page (the .png with the white background, attached) has 4 dash
lines per segment whereas my solution has 5 (.png with the black
background, attached). And as they say, "five is right out!"

Thank you,
Eowyn

**Comment by PythonCHB** [comment]
Ah -- I think that's 'cause the problem is not precisely specified -- there are multiple options for rounding, etc.

I think your solution is fine :-)

**Pull request merged**